<p>Feature files should not contain multiple consecutive blank lines.
A single blank line is sufficient to separate structural elements (e.g., between
scenarios or between a background and a scenario). Two or more consecutive blank
lines add unnecessary visual noise and can indicate sloppy formatting.</p>

<h2>Noncompliant Code Example</h2>
<pre>
Feature: Order management

  Scenario: Create order
    Given a customer account
    When the customer places an order
    Then the order is created


  <!-- Noncompliant: second consecutive blank line -->
  Scenario: Cancel order
    Given an existing order
    When the customer cancels the order
    Then the order is cancelled
</pre>

<h2>Compliant Solution</h2>
<pre>
Feature: Order management

  Scenario: Create order
    Given a customer account
    When the customer places an order
    Then the order is created

  Scenario: Cancel order
    Given an existing order
    When the customer cancels the order
    Then the order is cancelled
</pre>

<h2>See Also</h2>
<ul>
  <li><code>blank-line-before-scenario</code> â€” ensures a blank line precedes each scenario.</li>
</ul>
