<p>
  All columns in an <code>Examples</code> table should be referenced as
  <code>&lt;variable&gt;</code> placeholders in at least one step of the
  owning <code>Scenario Outline</code>. Unreferenced columns represent dead
  test data that add noise and can confuse readers about which values
  actually drive the test.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
Feature: User registration

  Scenario Outline: Register a new user
    Given the registration form is displayed
    When the user registers as a &lt;role&gt;
    Then the account is created

    Examples:                                &lt;!-- Noncompliant: "unused" column not referenced --&gt;
      | role    | unused   |
      | admin   | foo      |
      | editor  | bar      |
</pre>

<h2>Compliant Solution</h2>
<p>Remove unreferenced columns from the Examples table:</p>
<pre>
Feature: User registration

  Scenario Outline: Register a new user
    Given the registration form is displayed
    When the user registers as a &lt;role&gt;
    Then the account is created

    Examples:
      | role    |
      | admin   |
      | editor  |
</pre>

<h2>See Also</h2>
<ul>
  <li><code>examples-column-coverage</code> &mdash; Steps must not reference variables missing from the Examples table</li>
</ul>
