<p>
  Steps should not have an <code>UNKNOWN</code> keyword type. This typically
  occurs when the <code>*</code> (asterisk) prefix is used instead of an
  explicit keyword like <code>Given</code>, <code>When</code>,
  <code>Then</code>, <code>And</code>, or <code>But</code>. The parser
  cannot determine the semantic role of such steps, making the scenario
  harder to understand and analyse.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
Feature: Order management

  Scenario: Place an order
    Given the customer has an active account
    * the customer performs an action                    &lt;!-- Noncompliant --&gt;
    Then the order is placed successfully
</pre>

<h2>Compliant Solution</h2>
<pre>
Feature: Order management

  Scenario: Place an order
    Given the customer has an active account
    When the customer performs an action
    Then the order is placed successfully
</pre>

<h2>Notes</h2>
<p>
  This rule is related to <code>no-star-step-prefix</code> but focuses on
  the parsed keyword type rather than the literal prefix character. Both
  rules help enforce clear, explicit step classification.
</p>
