<p>Tag lines should have exactly one space between consecutive <code>@</code>-prefixed
tokens. Multiple spaces or tabs between tags make the tag line harder to read and
create inconsistent formatting across the feature file.</p>

<h2>Noncompliant Code Example</h2>
<pre>
Feature: Search functionality

  @smoke  @regression <!-- Noncompliant: two spaces between tags -->
  Scenario: Basic search
    Given the search page is loaded
    When the user searches for "test"
    Then results are displayed

  @smoke    @integration    @api <!-- Noncompliant: multiple spaces -->
  Scenario: Advanced search
    Given the search page is loaded
    When the user applies filters
    Then filtered results are displayed
</pre>

<h2>Compliant Solution</h2>
<pre>
Feature: Search functionality

  @smoke @regression
  Scenario: Basic search
    Given the search page is loaded
    When the user searches for "test"
    Then results are displayed

  @smoke @integration @api
  Scenario: Advanced search
    Given the search page is loaded
    When the user applies filters
    Then filtered results are displayed
</pre>

<h2>See Also</h2>
<ul>
  <li><code>no-trailing-whitespace</code> - removes trailing whitespace from all lines.</li>
  <li><code>tag-name-pattern</code> - validates tag naming conventions.</li>
</ul>
