<p>
  Feature files must be valid Gherkin that the parser can process without
  errors. When a file cannot be parsed, an issue is raised at the location
  of the parse error. This typically indicates a syntax mistake such as a
  missing keyword, misplaced table, or unrecognised structure.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
Feature: User login

  Scenario: Login with valid credentials
    the user has a valid account                         &lt;!-- Noncompliant: missing step keyword --&gt;
    When the user logs in
    Then the user sees the dashboard
</pre>

<h2>Compliant Solution</h2>
<pre>
Feature: User login

  Scenario: Login with valid credentials
    Given the user has a valid account
    When the user logs in
    Then the user sees the dashboard
</pre>

<h2>Notes</h2>
<p>
  This rule is a <em>marker</em> rule. Issues are created by the sensor
  when parsing fails &mdash; the check class itself does not visit the AST.
  Fixing the parse error in the feature file resolves the issue.
</p>
