<p>Every <code>Scenario</code> and <code>Scenario Outline</code> should have at least
one tag matching a configurable regular expression pattern. This rule is useful for
teams that require categorization tags (e.g., <code>@smoke</code>, <code>@regression</code>)
on every scenario to ensure proper test suite organization and selective execution.</p>

<h2>Noncompliant Code Example</h2>
<pre>
Feature: User login

  <!-- Noncompliant: scenario has no tags matching "smoke|regression" -->
  Scenario: Successful login
    Given a registered user
    When the user logs in with valid credentials
    Then the user sees the dashboard
</pre>

<h2>Compliant Solution</h2>
<pre>
Feature: User login

  @smoke
  Scenario: Successful login
    Given a registered user
    When the user logs in with valid credentials
    Then the user sees the dashboard
</pre>

<h2>Configuration</h2>
<table>
  <thead>
    <tr><th>Parameter</th><th>Description</th><th>Default</th></tr>
  </thead>
  <tbody>
    <tr>
      <td><code>pattern</code></td>
      <td>Regular expression that at least one tag must match (without the leading <code>@</code>).
          The default <code>.*</code> matches everything, so the rule is effectively inactive
          unless configured with a meaningful pattern.</td>
      <td><code>.*</code></td>
    </tr>
  </tbody>
</table>

<h2>See Also</h2>
<ul>
  <li><code>tag-name-pattern</code> — validates individual tag naming conventions.</li>
  <li><code>no-restricted-tags</code> — flags tags from a restricted list.</li>
</ul>
